src/App.tsx

```diff
@@ -7,33 +9,80 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  time: string;
+  timerName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    time: '',
+    timerName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerID = 0;
+
+  componentDidMount() {
+    this.timerID = window.setInterval(() => {
+      this.setState(({ timerName: getRandomName() }
+      ));
+    }, 3300);
+
+    window.setInterval(() => {
+      const time = new Date().toLocaleTimeString();
+
+      this.setState({ time });
+      console.info(time);
+    }, 1000);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('mousedown', this.clockShow);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentDidUpdate(prevState: State) {
+    if (this.state.hasClock === true
+      && this.state.timerName !== prevState.timerName) {
+      console.warn(`Renamed from ${prevState.timerName} to ${this.state.timerName}`);
```

Comment:
here should be console.debug. 
```suggestion
      console.debug(`Renamed from ${prevState.timerName} to ${this.state.timerName}`);
```

And the clock name can't be undefined. Check it one more time, please 
<img width="472" alt="image" src="https://user-images.githubusercontent.com/99403198/229272457-f2879567-f89b-4c6d-98aa-dfe71da2a21f.png">